lib/data.go

```diff
@@ -85,6 +85,49 @@ func (a Amount) Sub(sub Amount) (Amount, error) {
 	return a - sub, nil
 }
 
+//
+// Add this `Amount` to itself, `n` times
+//
+// If the resulting value would overflow maximumAmount, an error is returned,
+// along with the value (which would trigger a `panic` if used).
+//
+func (a Amount) Mult(n int) (Amount, error) {
+	return a.Mult64(int64(n))
+}
+
+/// Ditto
+func (a Amount) Multu(n uint) (Amount, error) {
+	return a.Multu64(uint64(n))
+}
+
+/// Ditto
+func (a Amount) Mult64(n int64) (Amount, error) {
+	if n < 0 {
+		return invalidValue, sebakerror.ErrorAccountBalanceUnderZero
+	}
+	return a.Multu64(uint64(n))
+}
+
+/// Ditto
+func (a Amount) Multu64(n uint64) (Amount, error) {
+	a.Invariant()
+	if uint64(MaximumBalance)/n < uint64(a) {
+		return invalidValue, sebakerror.ErrorMaximumBalanceReached
+	}
+
+	return Amount(uint64(a) * n), nil
+}
+
+// Counterpart of `Mult` which panic instead of returning an error
+// Useful for debugging and testing, should be avoided in regular code
+func (a Amount) MustMult(n int) Amount {
+	if v, err := a.Mult(n); err != nil {
+		panic(err)
+	} else {
+		return v
+	}
+}
+
```

Comment:
How about using function name like this for readability? It's not mandatory :)
```
MultiplyInt
MultiplyUint
MultiplyInt64
MultiplyUint64
MustMultiply
```

Reply:
Hum I attempted to keep the name of the operation shorts (e.g. with have `Sub` not `Subtract`). I'll add the types though.